src/ossl/aes.rs

```diff
@@ -2632,9 +2636,20 @@ impl AesCmacOperation {
         buf.zeroize();
 
         #[cfg(feature = "fips")]
-        {
-            self.fips_approved = check_mac_fips_indicators(&mut self.ctx)?;
-        }
+        self.fips_approval_cmac()?;
+
+        Ok(())
+    }
+
+    #[cfg(feature = "fips")]
+    fn fips_approval_cmac(&mut self) -> Result<()> {
+        /* The MAC len needs to have specific length for FIPS
+         * NIST SP 800-38B A.2:
+         * > For most applications, a value for Tlen that is at least 64
+         * should provide sufficient protection against guessing attacks.
+         */
+        self.fips_approved = Some(self.maclen >= (64 / 8));
```

Comment:
sounds like you use 64/8 to indicate we need to check for 64bit length but the quantity is in bytes.

I would see us either definintg a const or using:
   64 / sizeof!(u8) 

sizeof is a macro in src/misc.c

Reply:
ah but sizeof() returns the result in CK_ULONG, so I guess you'll have to do:
 64 / std::mem::size_of::<u8>()
instead

Reply:
Correct. I wanted to express the bytes/bits conversion here for readability. The standards talk about bits, but we have lengths in bytes.

The sizeof will return the number of bytes and not bits that I would need here. Getting number of bits from some constant (especially from u8) sounds to me quite redundant.

Moreover, I think everything in the world now counts on 8 bits in byte.

Reply:
bah I am not thinking straight yet :)

Maybe provide a macro?
bits_to_bytes!(64)

?

Reply:
That sounds to me over engineering the task of getting a number 8 from 64. Maybe some appropriate comment would server better?

Reply:
A comment and just the number 8 would be fine for this case.

Reply:
After trying to find at least some reference to put into the comment, I do not see anything specific in
https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf
It is still not clear to me if even the 64b should be approved or 32 should, as the wording is as vague as Stephan's:

> t may be any one of the following five values: 128, 120, 112, 104, or 96. For certain applications, t may be 64 or 32;

Reply:
Changed to plain numbers and added appropriate comments.